br/pkg/lightning/restore/precheck_impl.go

```diff
@@ -152,14 +166,26 @@ func (ci *clusterResourceCheckItem) Check(ctx context.Context) (*CheckResult, er
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	estimateSize := clusterSource * replicaCount
-	if estimateSize > clusterAvail {
+	estimateTikvSize := tikvSourceSize * replicaCount
+	// note: tiflashSourceSize contains replicaCount
+	estimateTifalshSize := tiflashSourceSize
+	if estimateTikvSize <= tikvAvail && estimateTifalshSize <= tiflashAvail {
+		theResult.Message = fmt.Sprintf("The storage space is rich, which TiKV/Tiflash is %s/%s. The estimated storage space is %s/%s.",
+			units.BytesSize(float64(tikvAvail)), units.BytesSize(float64(tiflashAvail)), units.BytesSize(float64(estimateTikvSize)), units.BytesSize(float64(estimateTifalshSize)))
+	}
+
+	if estimateTikvSize > tikvAvail {
 		theResult.Passed = false
-		theResult.Message = fmt.Sprintf("Cluster doesn't have enough space, available is %s, but we need %s",
-			units.BytesSize(float64(clusterAvail)), units.BytesSize(float64(estimateSize)))
-	} else {
-		theResult.Message = fmt.Sprintf("Cluster available is rich, available is %s, we need %s",
-			units.BytesSize(float64(clusterAvail)), units.BytesSize(float64(estimateSize)))
+		theResult.Message += fmt.Sprintf("The estimated storage space required by the TiKV cluster is %s but the actual storage space is %s.",
+			units.BytesSize(float64(tikvAvail)), units.BytesSize(float64(estimateTikvSize)))
```

Comment:
Is this order wrong?

Reply:
yes, fixed in https://github.com/pingcap/tidb/pull/41943/commits/2da4928b04bf19365090cc061d70ad778aa4d0f9